include/nuttx/mutex.h

```diff
@@ -113,7 +120,14 @@ static inline int nxmutex_init(FAR mutex_t *mutex)
 
 static inline int nxmutex_destroy(FAR mutex_t *mutex)
 {
-  return nxsem_destroy(mutex);
+  int ret = _SEM_DESTROY(mutex);
+
+  if (ret < 0)
+    {
+      return -(_SEM_ERRNO(ret));
```

Comment:
_SEM_ERRVAL(ret)

Reply:
Yes, but _SEM_ERRVAL(ret) return the negative value too, but more simpler.

Reply:
Let me check. Latest time I was looking into it I think it was returning positive value

Reply:
Yes. True. We have two of those:
```
#  define _SEM_ERRNO(r)         (-(r))
#  define _SEM_ERRVAL(r)        (r)
```
You are completely right, again!